src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/TdsParser.cs

```diff
@@ -8,6 +8,8 @@ namespace Microsoft.Data.SqlClient
 {
     internal partial class TdsParser
     {
+        private static readonly SqlObjectPool<ArrayBufferWriter<byte>> _writers = new(20, () => new(), a => a.Clear());
```

Comment:
No complain here only I'm curious about the constant 20 pooled objects here!
cc @David-Engel

Reply:
honestly I don't remember where the 20 came from. Seemed reasonable I think at the time, but I'm not really aware of how many concurrent calls to it are called in a given connection